core/src/main/scala/pureconfig/error/ConfigReaderFailure.scala

```diff
@@ -298,6 +299,16 @@ final case class NoValidCoproductChoiceFound(value: ConfigValue, location: Optio
 }
 
 /**
+ * A failure representing the inability to read a requested file.
+ */
+final case class CannotReadFile(path: Path) extends ConfigReaderFailure {
```

Comment:
Is it worth distinguishing between `path` not being a file vs not being able to read the file?